packages/core/src/modules/indy/repository/AnonCredsCredentialDefinitionRecord.ts

```diff
@@ -1,39 +1,47 @@
+import type { DidIndyNamespace } from '../../../utils/indyIdentifiers'
 import type { CredDef } from 'indy-sdk'
 
 import { BaseRecord } from '../../../storage/BaseRecord'
-import { didFromCredentialDefinitionId } from '../../../utils/did'
+import { getQualifiedIdentifier } from '../../../utils/indyIdentifiers'
+import { CredentialDefinitionTransformer } from '../../../utils/transformers'
+import { uuid } from '../../../utils/uuid'
 
 export interface AnonCredsCredentialDefinitionRecordProps {
   credentialDefinition: CredDef
+  didIndyNamespace: DidIndyNamespace
+  schemaSeqNo: number
 }
 
-export type DefaultAnonCredsCredentialDefinitionTags = {
-  credentialDefinitionId: string
-  issuerDid: string
-  schemaId: string
-  tag: string
-}
-
-export class AnonCredsCredentialDefinitionRecord extends BaseRecord<DefaultAnonCredsCredentialDefinitionTags> {
+export class AnonCredsCredentialDefinitionRecord extends BaseRecord {
   public static readonly type = 'AnonCredsCredentialDefinitionRecord'
   public readonly type = AnonCredsCredentialDefinitionRecord.type
-  public readonly credentialDefinition!: CredDef
+
+  @CredentialDefinitionTransformer()
+  public readonly credentialDefinition!: CredDef & { didIndyNamespace?: DidIndyNamespace; schemaSeqNo?: number }
+
+  public readonly schemaSeqNo!: number
 
   public constructor(props: AnonCredsCredentialDefinitionRecordProps) {
     super()
 
+    this.id = uuid()
     if (props) {
-      this.credentialDefinition = props.credentialDefinition
+      this.credentialDefinition = {
+        ...props.credentialDefinition,
+        didIndyNamespace: props.didIndyNamespace,
+        schemaSeqNo: props.schemaSeqNo,
+      }
+      this.schemaSeqNo = props.schemaSeqNo
+      this._tags.id = getQualifiedIdentifier(props.didIndyNamespace, {
+        ...this.credentialDefinition,
+        schemaSeqNo: props.schemaSeqNo,
+      })
```

Comment:
I think it is better to do the transformation outside of the record class as the record will be used by other anoncreds services.


```suggestion
      this.credentialDefinition = props.credentialDefinition
```